src/appgateway.tf

```diff
@@ -53,30 +53,40 @@ module "app_gw" {
   # Configure backends
   backends = {
     apim = {
-      protocol     = "Https"
-      host         = trim(azurerm_dns_a_record.dns_a_api.fqdn, ".")
-      port         = 443
-      ip_addresses = module.apim.private_ip_addresses
-      probe        = "/status-0123456789abcdef"
-      probe_name   = "probe-apim"
+      protocol                    = "Https"
+      host                        = trim(azurerm_dns_a_record.dns_a_api.fqdn, ".")
+      port                        = 443
+      ip_addresses                = module.apim.private_ip_addresses
+      fqdns                       = [azurerm_dns_a_record.dns_a_api.fqdn]
+      probe                       = "/status-0123456789abcdef"
+      probe_name                  = "probe-apim"
+      request_timeout             = 180
```

Comment:
@infantesimone how much is your expected timeout? probably 180 seconds are too high. For IO app we have 10-8 seconds